### ISSUE TITLE
Index has error info while insert vertex

### ISSUE THREAD

**Issue by dataweng** [orientechnologies/orientdb#7351]
## Expected behavior  
insert vertex

## Actual behavior  

```
2017-04-17 18:56:01:021 WARNI {db=gacenter_tst} Page in file certification_3.pcl with index 4650 was placed in wrong free list, this error will be fixed automatically [OPaginatedCluster]
2017-04-17 18:56:01:075 WARNI {db=gacenter_tst} Page in file certification_3.pcl with index 4650 was placed in wrong free list, this error will be fixed automatically [OPaginatedCluster]$ANSI{green {db=gacenter_tst}} Error on creating record in cluster: plocal cluster: certification_3
java.lang.NullPointerException
$ANSI{green {db=gacenter_tst}} Error on creating record in cluster: plocal cluster: certification_3
com.orientechnologies.orient.core.exception.OPaginatedClusterException: Error during record creation^M
        DB name="gacenter_tst"^M
        Component Name="certification_3"
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OPaginatedCluster.createSinglePageRecord(OPaginatedCluster.java:609)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OPaginatedCluster.createDataRecord(OPaginatedCluster.java:486)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OPaginatedCluster.createRecord(OPaginatedCluster.java:451)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doCreateRecord(OAbstractPaginatedStorage.java:3032)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.createRecord(OAbstractPaginatedStorage.java:1069)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeSaveRecord(ODatabaseDocumentTx.java:2178)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveNew(OTransactionNoTx.java:241)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveRecord(OTransactionNoTx.java:171)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:2707)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:102)
        at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1805)
        at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1796)
        at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:63)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLInsert.saveRecord(OCommandExecutorSQLInsert.java:330)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLInsert.execute(OCommandExecutorSQLInsert.java:236)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.java:74)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.executeCommand(OAbstractPaginatedStorage.java:2615)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.command(OAbstractPaginatedStorage.java:2561)
        at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract.execute(OCommandRequestTextAbstract.java:69)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.command(ONetworkProtocolBinary.java:1473)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:597)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.sessionRequest(ONetworkProtocolBinary.java:336)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.execute(ONetworkProtocolBinary.java:200)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
Caused by: java.lang.NullPointerException
```

## Steps to reproduce  
insert into certification(c_id, name, create_time, update_time) values('1234567','test',1480911729, 1480911730)

**Comment by luigidellaquila** [comment]
Hi @dataweng 

Do you have a test case, an existing db or a script to reproduce the problem?

Thanks

**Comment by laa** [comment]
@dataweng also I have a question about this database. Was it created under 2.2.18 version or that is earlier version of data ?

**Comment by dataweng** [comment]
db created by 2.2.18，but lib is orientdb-jdbc-2.2.13.jar

**Comment by laa** [comment]
@dataweng could you send server logs to analyze?

**Comment by dataweng** [comment]
this is server log. tks!

**Comment by laa** [comment]
@dataweng I suppose you wanted to send me a link. Could you provide me more details about this issue? How do you get it? Maybe you caught OOM or some other exceptions? Could you check server directory? Maybe you have .hprof files or JVM error reports ?

**Comment by dataweng** [comment]
create class certification if not exists extends V
create class relation if not exists extends E
alter class certification shortname C
alter class relation shortname R

create property certification.c_id if not exists String
create property certification.name if not exists String
create property certification.create_time if not exists DateTime
create property certification.update_time if not exists DateTime
 
create property relation.r_id if not exists String
create property relation.create_time if not exists DateTime
create property relation.update_time if not exists DateTime

create index index_cert on certification(c_id) UNIQUE

**Comment by dataweng** [comment]
GitHub do not pass up the attachment, this is my creation statement, now is the phenomenon of storage of about 25 million, there is such a situation, this is a stand-alone mode, however, just found before storage procedures use the 2.2.13 JAR package, should be in the wrong, and now into the 2.2.18JAR package, night run look at the results tomorrow.

**Comment by dataweng** [comment]
In this case, the program can continue to enter the vertices and edges, but the speed is super slow, even only a few seconds, my program and OServerMain have been running, did not hang up, it is strange. Whether or not the content of this field is related to c_id, because the content of c_id may be garbled.

**Comment by laa** [comment]
@dataweng your feedback is not completely clear for me. So I will describe you what I thought about it. In versions before 2.2.17, we got an issue when some data from the cluster may be truncated, and your exception reminds me exception which is thrown in such case. Is it possible that you initially operate on the database under version less than 2.2.17 and then switched to a newer version? In such case I recommend you to execute import/export of database from to json and run the same tests on 2.2.18 version. 

About server logs do you have google drive or something so you attach the link and I will be able to download them ?

**Comment by dataweng** [comment]
https://drive.google.com/file/d/0B5JJf0QzwRejOHFuZ1JJeHU1UVE/view
this is server logs url.

**Comment by laa** [comment]
Hi @dataweng that is a log from the latest run, in a nutshell, it repeats exception which you sent to us. But we need the log for the last seven days to check what precedes this issue. It is OK if you send us zip file. Could you also check that whether you have "have .hprof files or JVM error reports" ? Could you also explain what did you mean when you wrote that "found before storage procedures use the 2.2.13 JAR package, should be in the wrong, and now into the 2.2.18JAR package." it is not clear for me.

**Comment by dataweng** [comment]
I mean: in my application package for JAR or 2.2.13, this is yesterday that the improper use of the problem, but last night in my application on JAR packets into 2.2.18, are having the same problem, but the orientdb error log content may be slightly different, I will also send you to see.

**Comment by dataweng** [comment]
JVM error, I reconfirm. Orientdb log is only one day or half a day, because it can not run so long on the wrong, I am now through my application to import hundreds of millions of vertices and edges, but in the storage to tens of millions of errors. Lead to work is difficult to carry out, my application has been unable to on-line.

**Comment by dataweng** [comment]
/appslog/orientdb/orientdb-enterprise-2.2.18 directory, there is no.Hprof file, and JVM did not make a mistake. I describe my application using orientdb: my application is spark streaming, in streaming, JDBC (orientdb-jdbc-2.2.18.jar) prepared by the SQL statement to access orientdb, only insert and update. Throughout the gacenter_tst library, there are only two tables certification and relation. Server configuration: 128G/32cores/raid5

**Comment by dataweng** [comment]
https://drive.google.com/open?id=0B5JJf0QzwRejR0p0Yi1jQWpoYU0
this is server log for orientdb-jdbc-2.2.18.jar.

**Comment by dataweng** [comment]
if [ -z "$ORIENTDB_OPTS_MEMORY" ] ; then
    ORIENTDB_OPTS_MEMORY="**_**-Xms20G -Xmx20G***_*"
fi

if [ -z "$JAVA_OPTS_SCRIPT" ] ; then
    JAVA_OPTS_SCRIPT="-Djna.nosys=true -XX:+HeapDumpOnOutOfMemoryError -**_**XX:MaxDirectMemorySize=30g**_** -Djava.awt.headless=true -Dfile.encoding=UTF8 -Drhino.opt.level=9"
fi

**Comment by laa** [comment]
@dataweng I looked at the log, could you confirm, so  during the test you create database, run the test on new database and you got this exception , is it correct ?

**Comment by dataweng** [comment]
Yes, because it is still in the test phase, the database is re created, and every time I run a few hours of this situation, the vertex is probably about 30 million into the abnormal.

**Comment by laa** [comment]
@dataweng this issue is the highest priority for us do you have chat (hangout/skype/slack) to contact with you? We will also send you patched version of the project to understand what is the cause of the issue.

**Comment by dataweng** [comment]
==

**Comment by laa** [comment]
@dataweng what do you mean ?

**Comment by dataweng** [comment]
waiting

**Comment by laa** [comment]
@dataweng do you have chat to contact with you asap ?

**Comment by dataweng** [comment]
no, skype

**Comment by dataweng** [comment]
pls give me a skype account.

**Comment by laa** [comment]
That is skype of developer who is working on your issue now actual.serge

**Comment by dataweng** [comment]
Thank you very much for your support!

**Comment by taburet** [comment]
@dataweng do you still experiencing the problems? can we close the issue?

**Comment by laa** [comment]
I have tried following approach, I have created a database with about 100 million of records under the load A of YCSB tests.  Then I have been running Load A during the updates length of the new fields was uniformly distributed. Load A was running during a week continuously, but I was not able to reproduce either exception or even "Page in file certification_3.pcl with index 4650 was placed in a wrong free list, this error will be fixed automatically " message. Will try to write a different kind of test soon.
